templates/swift/Sources/Services/Realtime.swift.twig

```diff
@@ -98,36 +110,38 @@ open class Realtime : Service {
         channels: Set<String>,
         payloadType: T.Type,
         callback: @escaping (RealtimeResponseEvent) -> Void
-    ) -> RealtimeSubscription {
+    ) async throws -> RealtimeSubscription {
         subscriptionsCounter += 1
-        let counter = subscriptionsCounter
+
+        let count = subscriptionsCounter
 
         channels.forEach {
             activeChannels.insert($0)
         }
 
-        activeSubscriptions[counter] = RealtimeCallback(
+        activeSubscriptions[count] = RealtimeCallback(
             for: Set(channels),
-            and: callback
+            with: callback
         )
 
         connectSync.sync {
             subCallDepth+=1
         }
 
-        DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(DEBOUNCE_MILLIS)) {
-            if (self.subCallDepth == 1) {
-                self.createSocket()
-            }
-            self.connectSync.sync {
-                self.subCallDepth-=1
-            }
+        try await Task.sleep(nanoseconds: UInt64(DEBOUNCE_NANOS))
+
+        if self.subCallDepth == 1 {
+            try await self.createSocket()
+        }
+
+        connectSync.sync {
+            self.subCallDepth -= 1
         }
 
         return RealtimeSubscription {
-            self.activeSubscriptions[counter] = nil
+            self.activeSubscriptions[count] = nil
```

Comment:
This too troubles me, we use active subscription to send back callback (https://github.com/appwrite/sdk-generator/blob/f67b1a4b953e82d32cfe842452f842867711c36d/templates/swift/Sources/Services/Realtime.swift.twig#L218-L219)

Reply:
This is expected, the lambda passed to the `RealtimeSubscription` is the `close` function. So when close is called, we remove it from the activeSubscriptions